cmd/view-fits/main.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015 The astrogo Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"log"
+	"os"
+
+	"github.com/astrogo/fitsio"
+
+	"golang.org/x/exp/shiny/driver"
+	"golang.org/x/exp/shiny/screen"
+	"golang.org/x/mobile/event/key"
+	"golang.org/x/mobile/event/paint"
+)
+
+var (
+	hduArray []fitsio.Image
+)
+
+func init() {
+
+	var inputFileName string
+	var inputFile *os.File
+	var fitsFile *fitsio.File
+	var err error
+
+	if len(os.Args) < 2 {
+		log.Fatal("no image file specified")
+	}
+
+	// Getting the file argument.
+	inputFileName = os.Args[1]
+
+	// Opening the file.
+	if inputFile, err = os.Open(inputFileName); err != nil {
+		log.Fatal("Can not open the input file: %s.", err)
+	}
+
+	// Opening the FITS file.
+	if fitsFile, err = fitsio.Open(inputFile); err != nil {
```

Comment:
the `fitsFile` should be closed (`defer f.Close()`)once the processing is done, otherwise we leak file descriptors resources.